src/ui/drawing/gdi/GDIBitmapSurface.hh

```diff
@@ -0,0 +1,122 @@
+#ifndef RA_UI_DRAWING_GDI_BITMAP_SURFACE_H
+#define RA_UI_DRAWING_GDI_BITMAP_SURFACE_H
+#pragma once
+
+#include "RA_Core.h"
+
+#include "ui\drawing\gdi\GDISurface.hh"
+#include "ui\drawing\gdi\ResourceRepository.hh"
+
+namespace ra {
+namespace ui {
+namespace drawing {
+namespace gdi {
+
+class GDIBitmapSurface : public GDISurface
+{
+public:
+    explicit GDIBitmapSurface(const int nWidth, const int nHeight, ResourceRepository& pResourceRepository) noexcept
+        : GDISurface(CreateBitmapHDC(nWidth, nHeight), RECT{ 0, 0, nWidth, nHeight }, pResourceRepository)
+    {
+    }
+
+    explicit GDIBitmapSurface(const int nWidth, const int nHeight) noexcept
+        : GDISurface(CreateBitmapHDC(nWidth, nHeight), RECT { 0, 0, nWidth, nHeight })
+    {
+    }
+
+    GDIBitmapSurface(const GDIBitmapSurface&) noexcept = delete;
+    GDIBitmapSurface& operator=(const GDIBitmapSurface&) noexcept = delete;
+    GDIBitmapSurface(GDIBitmapSurface&&) noexcept = delete;
+    GDIBitmapSurface& operator=(GDIBitmapSurface&&) noexcept = delete;
+
+    ~GDIBitmapSurface()
```

Comment:
Destructors need to be `noexcept` no matter what.

Reply:
ok